# **In-Silico Structural and Functional Analysis of Cytochrome P450 Enzyme in *Aedes aegypti***

This repository contains the computational modeling and active-site prediction of a Cytochrome P450 (CYP450) enzyme from *Aedes aegypti*, a mosquito species which is a primary vector of dengue, zika and chikungunya virus.
The project integrates multiple bioinformatics approaches including sequence retrieval, homology modeling, and ligand-binding site prediction,to explore the enzyme's structure and function. The purpose of this project is to help uncover the mechanism of insecticide resistance in CYP450, a detoxifying enzyme which poses a major challenge in vector control strategies.
## Project Report
[View full report (PDF)] https://github.com/phil22nengminza/InSilico-Analysis-CYP450-/blob/3f8ef8b73ec153987341cb8b2c642444e7185294/In-Silico%20Structural%20and%20Functional%20Analysis%20of%20Cytochrome%20P450%20Enzyme%20in%20Aedes%20aegypti.pdf.pdf

## Tools & Databases used
**UniProt** - Protein sequence retrieval  

**SWISS-MODEL** - Homology modeling

**PrankWeb (P2Rank)** - Ligand-binding site prediction

##  Summary
The selected Cytochrome P450 model displayed the conserved P450 fold, dominated by α-helices surrounding a central cavity.  
PrankWeb predicted several potential ligand-binding pockets, **among which pocket 3 stood out to have major significance in catalytic function due to its higher average conservation.**
## Figures
### **Figure 1. Predicted 3D Model** 
[Model Overview] https://github.com/phil22nengminza/InSilico-Analysis-CYP450-/blob/63483de38586189d508fbf261f9deefd5ddd8e55/_API_V2_PREDICTION_V4-USER-UPLOAD_2025-10-31-15-52-55-15E6EBF9-0042-4242-89FE-C58328A324BA_PUBLIC_STRUCTURE.PDB.png
### **Figure 2. Predicted Pocket 1**
[Pocket 1] https://github.com/phil22nengminza/InSilico-Analysis-CYP450-/blob/7718508c941785603f3f1309f88e5b11fbe5d193/_API_V2_PREDICTION_V4-USER-UPLOAD_2025-10-31-15-52-55-15E6EBF9-0042-4242-89FE-C58328A324BA_PUBLIC_STRUCTURE.PDB%20(1).png

### **Figure 3. Predicted Pocket 2**
[Pocket 2] https://github.com/phil22nengminza/InSilico-Analysis-CYP450-/blob/946afed541c516c1fca31719ba865e75d4f61cac/_API_V2_PREDICTION_V4-USER-UPLOAD_2025-11-07-04-57-16-05C52716-023D-486A-97B6-3A7E1936B04B_PUBLIC_STRUCTURE.PDB%20(2).png

### **Figure 4. Predicted Pocket 3**
[Pocket 3] https://github.com/phil22nengminza/InSilico-Analysis-CYP450-/blob/b1ac48efc58a9111b1c91042f1298629f7ef8eb8/_API_V2_PREDICTION_V4-USER-UPLOAD_2025-11-07-04-57-16-05C52716-023D-486A-97B6-3A7E1936B04B_PUBLIC_STRUCTURE.PDB.png
## References
The UniProt Consortium , UniProt: the Universal Protein Knowledgebase in 2025. (2025). Nucleic Acids Research, 53(D1).https://doi.org/10.1093/nar/gkae1010  

 

Waterhouse, A., Bertoni, M., Bienert, S., Studer, G., Tauriello, G., Gumienny, R., Heer, F. T., de Beer, T. A. P., Rempfer, C., Bordoli, L., Lepore, R., & Schwede, T. (2018). SWISS-MODEL: Homology modelling of protein structures and complexes. Nucleic Acids Research, 46(W1). https://doi.org/10.1093/nar/gky427 

 

Bienert, S., Waterhouse, A., de Beer, T. A. P., Tauriello, G., Studer, G., Bordoli, L., & Schwede, T. (2017). The SWISS-MODEL Repository-new features and functionality. Nucleic Acids Research, 45(D1). https://doi.org/10.1093/nar/gkw1132 

 

Guex, N., Peitsch, M. C., & Schwede, T. (2009). Automated comparative protein structure modeling with SWISS-MODEL and Swiss-PdbViewer: A historical perspective. Electrophoresis, 30(SUPPL. 1). https://doi.org/10.1002/elps.200900140 

 

Studer, G., Rempfer, C., Waterhouse, A. M., Gumienny, R., Haas, J., & Schwede, T. (2020). QMEANDisCo—distance constraints applied on model quality estimation. Bioinformatics, 36(6). https://doi.org/10.1093/bioinformatics/btz828 

 

Bertoni, M., Kiefer, F., Biasini, M., Bordoli, L., & Schwede, T. (2017). Modeling protein quaternary structure of homo- and hetero-oligomers beyond binary interactions by homology. Scientific Reports, 7(1). https://doi.org/10.1038/s41598-017-09654-8 

 

 Polak, L., Skoda, P., Riedlova, K., Krivak, R., Novotny, M.,  Hoksza, D. (2025). PrankWeb 4: a modular web server for protein–ligand binding site prediction and downstream analysis. Nucleic Acids Research, 53(W1).https://doi.org/10.1093/nar/gkaf421 

 
